src/core/core/service_engine.cpp

```diff
@@ -159,6 +160,7 @@ void service_node::get_queue_info(
 
 rpc_request_task *service_node::generate_intercepted_request_task(message_ex *req)
 {
+    req->tracer = make_unique<dsn::tool::latency_tracer>(req->header->id, "client_request");
```

Comment:
I think it's better to introduce the trace util first, and apply it in another PR after the util is well designed.
It will make PRs more independent, and will also reduce you workload.

Reply:
Yeah, it should be splitted